cupy/_misc/memory_ranges.py

```diff
@@ -28,6 +28,8 @@ def _get_memory_ptrs(x):
 
 
 def shares_memory(a, b, max_work=None):
+    if a is b:
```

Comment:
this seems to be breaking some tests when it should't :/

Reply:
I can take a look, but in the worst case, I think this was just a micro-optimization that can be reverted.